src/internal/connector/exchange/service_restore.go

```diff
@@ -649,10 +649,17 @@ func establishContactsRestoreLocation(
 		return cached, nil
 	}
 
+	folders[0] = "Corso_Restore_02-May-2023_00:06:19"
 	ctx = clues.Add(ctx, "is_new_cache", isNewCache)
 
+	// This is where destination folder gets created by corso in graph
+	// This is the POST operation which may fail if the folder exists
```

Comment:
technically not true, since you're handling this case, yeah?

Reply:
Lol, these were just some pointers I added for myself.